package/generateScss.mjs

```diff
@@ -58,20 +59,69 @@ const generateSCSSModule = async (moduleName, importObj) => {
       }
       generatedScss += `${key}: ${value};\n`;
     });
-    
+  
+  // media.scss
   } else if (moduleName.toLowerCase() === 'media') {
     Object.keys(importObj).forEach((queryName) => {
       const processedQuery = customMediaHelper.process(queryName);
       queryName = queryName.replace('--', '$');
       generatedScss += `${queryName}: '${processedQuery}';\n`;
     });
+  
+  // shadows.scss
+  } else if (moduleName.toLowerCase() === 'shadows') {
+    
+    let mapKeysValues = '';
+    const lightColor = Shadows['--shadow-color'];
+    const lightStrength = Shadows['--shadow-strength'];
+    const darkColor = Shadows['--shadow-color-@media:dark'];
+    const darkStrength = Shadows['--shadow-strength-@media:dark'];
+    const entries = Object.entries(importObj);
+    
+    for (let index = 0; index < entries.length; index++) {
+      let [key, value] = entries[index];
+      
+      if (key == '--shadow-color' || key == '--shadow-strength' || key.includes('@')) {
+        continue; // skip light and dark for the other loops
+      } 
+
+      key = key.replace('--shadow-', '');
+      if (key.includes('--inner-shadow-')) {
+        key = key.replace('--inner-shadow-', '\'inner-');
+        key = key.replace(/$/, '\'');
+      }
+     
+      value = value.replace(/var\(--(.*?)\)/g, '$$--$1');
+      value = value.replace(/hsl/g, 'Hsl')
+      mapKeysValues += `${key}: (${value})`;
+      
+      if (index < entries.length - 1) {
+        mapKeysValues += ',\n '; // Add comma and new line for all entries except the last one
+      }
+    };
     
+    generatedScss += `@use 'sass:map';
+
+@function shadow($level, $theme: light, $shadow-color: null, $shadow-strength: null) {
+  $--shadow-color: $shadow-color or if($theme == dark, ${darkColor}, if($theme == cssvar, var(--shadow-color), ${lightColor}));
```

Comment:
hold on, could you not add the css var? it was perfect without it.

three reasons:
- it increases the api surface and complexity for no benefit (it is still possible to pass custom property names without this code)
- custom property names are global and can unexpectedly clash.
- as I mentioned previously, it is easy to create an abstraction over it for convenience:
  ```scss
  @function custom-shadow($level) {
    @return op.shadow($level, $shadow-color: var(--shadow-color), $shadow-strength: var(--shadow-strength));
  }
  ```

Reply:
👍 I’ve removed the cssvar from the if statement. I think that’s just my personal preferences coming through to write less even if it adds complexity.